x/stakeibc/keeper/icacallbacks_delegate_test.go

```diff
@@ -0,0 +1,270 @@
+		ChainId:        chainId,
+		HostDenom:      atom,
+		IBCDenom:       ibcAtom,
+		RedemptionRate: sdk.NewDec(1.0),
+		Validators:		[]*types.Validator{&val1, &val2},
+		StakedBal: 		stakedBal,
+	}
+	depositRecord := recordtypes.DepositRecord{
+		Id:                 1,
+		DepositEpochNumber: 1,
+		HostZoneId:         chainId,
+		Amount:             balanceToStake,
+		Status:				recordtypes.DepositRecord_STAKE,
+	}
+	s.App.StakeibcKeeper.SetHostZone(s.Ctx(), hostZone)
+	s.App.RecordsKeeper.SetDepositRecord(s.Ctx(), depositRecord)
+	
+	packet := channeltypes.Packet{}
+	var msgs []sdk.Msg
+	msgs = append(msgs, &stakingTypes.MsgDelegate{}, &stakingTypes.MsgDelegate{})
+	ack := s.ICAPacketAcknowledgement(msgs)
+	val1SplitDelegation := types.SplitDelegation{
+		Validator: 		val1.Address,
+		Amount:			uint64(val1RelAmt),
+	}
+	val2SplitDelegation := types.SplitDelegation{
+		Validator: 		val2.Address,
+		Amount:			uint64(val2RelAmt),
+	}
+	callbackArgs := types.DelegateCallback{
+		HostZoneId:				chainId,
+		DepositRecordId:		depositRecord.Id,
+		SplitDelegations:		[]*types.SplitDelegation{&val1SplitDelegation, &val2SplitDelegation},
+	}
+	args, err := s.App.StakeibcKeeper.MarshalDelegateCallbackArgs(s.Ctx(), callbackArgs)
+	s.Require().NoError(err)
+
+	return DelegateCallbackTestCase{
+		initialState: DelegateCallbackState{
+			stakedBal:		stakedBal,
+			balanceToStake: balanceToStake,
+			depositRecord:	depositRecord,
+			callbackArgs:	callbackArgs,
+			val1Bal: 		val1Bal,
+			val2Bal: 		val2Bal,
+			val1RelAmt:		val1RelAmt,
+			val2RelAmt:		val2RelAmt,
+		},
+		validArgs: DelegateCallbackArgs{
+			packet:   packet,
+			ack: 	  &ack,
+			args:     args,
+		},
+	}
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_Successful() {
+	tc := s.SetupDelegateCallback()
+	initialState := tc.initialState
+	validArgs := tc.validArgs
+
+	err := stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, validArgs.args)
+	s.Require().NoError(err)
+
+	// Confirm stakedBal has increased
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), chainId)
+	s.Require().True(found)
+	s.Require().Equal(int64(initialState.stakedBal) + initialState.balanceToStake, int64(hostZone.StakedBal), "stakedBal should have increased")
+
+	// Confirm delegations have been added to validators
+	val1 := hostZone.Validators[0]
+	val2 := hostZone.Validators[1]
+	s.Require().Equal(int64(initialState.val1Bal) + initialState.val1RelAmt, int64(val1.DelegationAmt), "val1 balance should have increased")
+	s.Require().Equal(int64(initialState.val2Bal) + initialState.val2RelAmt, int64(val2.DelegationAmt), "val2 balance should have increased")
+	
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 0, "number of deposit records")
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_DelegateCallbackTimeout() {
+	tc := s.SetupDelegateCallback()
+	validArgs := tc.validArgs
```

Comment:
nit: sometimes it's a little more clear to name this `invalidArgs` in the failure cases

Reply:
good call! done